src/solver/modeler/api/include/antares/solver/modeler/api/linearProblemFiller.h

```diff
@@ -30,9 +32,19 @@ namespace Antares::Solver::Modeler::Api
 class LinearProblemFiller
 {
 public:
-    virtual void addVariables(LinearProblem* problem, LinearProblemData* data) = 0;
-    virtual void addConstraints(LinearProblem* problem, LinearProblemData* data) = 0;
-    virtual void addObjectiveCoefficients(LinearProblem* problem, LinearProblemData* data) = 0;
+    explicit LinearProblemFiller(std::shared_ptr<ILinearProblem> pb, LinearProblemData& data):
+        linearProblem_(pb),
+        LPdata_(data)
+    {
+    }
+
+    virtual void addVariables() = 0;
+    virtual void addConstraints() = 0;
+    virtual void addObjective() = 0;
+
+protected:
```

Comment:
For now, all LP fillers should keep a reference to the LP (as well as to LP data).
To me, the best way so that they all contain such references is to have their base class containing it.
According to Sonar, **protected** data members are usually a bad practice, because it introduces too much coupling between a class and its children.
However, when a class hierarchy is only 2 leveled, and meant to use polymorphism, like here, I guess hat this way to have common data members among classes is not troublesome.

Reply:
Class **LinearProblemFiller** has no longer data member.